app/controllers/account/tasks_controller.rb

```diff
@@ -1,47 +1,54 @@
 class Account::TasksController < Account::AccountController
   def show
     @project = parent
-    @task = @project.tasks.find(params[:id])
+    @task = parent.tasks.find(params[:id])
   end
 
   def new
     @project = parent
-    @task = @project.tasks.build(section_params)
+    @task = parent.tasks.build(section_params)
   end
 
   def create
+    @project = parent
     @task = parent.tasks.build(tasks_params)
-
     if @task.save
-      redirect_to account_workspace_project_path(parent.workspace_id, parent)
+      redirect_to account_project_task_path(parent, @task)
     else
       render :new
     end
   end
 
   def edit
     @project = parent
-    @task = @project.tasks.find(params[:id])
+    @task = parent.tasks.find(params[:id])
```

Comment:
why you calling again parent?

you have @project already

revert all changes